framework/meta/src/cli_args/meta_cli_args.rs

```diff
@@ -0,0 +1,61 @@
+use super::BuildArgs;
+
+#[derive(Default)]
+pub struct CliArgs {
+    pub action: CliAction,
+    pub load_abi_git_version: bool,
+}
+
+impl CliArgs {
+    pub fn parse(args: &[String]) -> Self {
+        let no_abi_git_version = args.iter().any(|arg| arg == "--no-abi-git-version");
+        CliArgs {
+            action: CliAction::parse(args),
+            load_abi_git_version: !no_abi_git_version,
+        }
+    }
+}
+
+pub enum CliAction {
+    Build(BuildArgs),
+    Clean,
+    GenerateSnippets(GenerateSnippetsArgs),
+    Nothing,
+}
```

Comment:
Defaults like `Nothing` are usually the first, which also lets you `#[derive(Default)]`. Any particular reason why it's not like that here?